---
title: "ERA Dada2 Pipeline"
output:
  html_document: default
  html_notebook: default
---



# Dada2 sequence pipeline for ERA Rhizosphere 
# Quality Control and Sequence Binning with DADA2 Multithreading Pipeline
* study, v4 Bacteria, library1, run 1
* Following Spencer's blog walkthrough https://spencerjd.github.io/DADA2-Walkthrough/

## Check Sequence Quality
```{r}
library(dada2)
```

### load file names

```{r}
path = "/home/bryan/ERA/data/MiSeq/20170417_run1/v4bac_lib1_demult/"
fns = list.files(path)

fastqs = fns[grepl(".fq.gz$", fns)]
fastqs = sort(fastqs) # Sort ensures forward/reverse reads are in same order
fnFs = fastqs[grepl("R1", fastqs)] # Just the forward read files
fnRs = fastqs[grepl("R2", fastqs)] # Just the reverse read files

# Get sample names from the first part of the forward read filenames
sample_names = sapply(strsplit(fnFs, "[.]"), `[`, 1)

# Fully specify the path for the fnFs and fnRs
fnFs <- paste0(path, fnFs)
fnRs <- paste0(path, fnRs)
```

```{r}
# Plot quality profile for each sample. Change the number in brackets to look at a different sample's profile.
plotQualityProfile(fnFs[[6]])
```

```{r}
# Plot quality profile for each sample. Change the number in brackets to look at a different sample's profile.
plotQualityProfile(fnRs[[6]])
```

```{r}
filtFs = paste0(path, sample_names, "_F_filt.fastq.gz")
filtRs = paste0(path, sample_names, "_R_filt.fastq.gz")

if(length(fnFs) != length(fnRs)) stop("Forward and reverse files do not match.")
# Filtering: THESE PARAMETERS ARENT OPTIMAL FOR ALL DATASETS. Trim sequences based on quality profile from raw fastq files. 
for(i in seq_along(fnFs)) {
  fastqPairedFilter(c(fnFs[i], fnRs[i]), c(filtFs[i], filtRs[i]),
                    trimLeft=c(10, 5), truncLen=c(155,125), 
                    maxN=0, maxEE=2, truncQ=2, 
                    compress=TRUE, verbose=TRUE)
}
```

## Run sample inference
```{r}
#Set sample names based on file names.
#Change split to match naming scheme
sample.names <- sapply(strsplit(basename(filtFs), "_F"), `[`, 1)
names(filtFs) <- sample.names
names(filtRs) <- sample.names
set.seed(100)
```

```{r}
# Learn forward error rates
NSAM.LEARN <- 50 # Choose enough samples to have at least 1M reads
drp.learnF <- derepFastq(sample(filtFs, NSAM.LEARN))
dd.learnF <- dada(drp.learnF, err=NULL, selfConsist=TRUE, multithread=TRUE)
errF <- dd.learnF[[1]]$err_out
rm(drp.learnF);rm(dd.learnF)
```

```{r}
# Learn reverse error rates
drp.learnR <- derepFastq(sample(filtRs, NSAM.LEARN))
dd.learnR <- dada(drp.learnR, err=NULL, selfConsist=TRUE, multithread=TRUE)
errR <- dd.learnR[[1]]$err_out
rm(drp.learnR);rm(dd.learnR)
```


```{r}
# Sample inference and merger of paired-end reads
mergers <- vector("list", length(sample.names))
names(mergers) <- sample.names
for(sam in sample.names) {
  cat("Processing:", sam, "\n")
    derepF <- derepFastq(filtFs[[sam]])
    ddF <- dada(derepF, err=errF, multithread=TRUE)
    derepR <- derepFastq(filtRs[[sam]])
    ddR <- dada(derepR, err=errR, multithread=TRUE)
    merger <- mergePairs(ddF, derepF, ddR, derepR)
    mergers[[sam]] <- merger
}
rm(derepF); rm(derepR)
# Construct sequence table and remove chimeras
seqtab <- makeSequenceTable(mergers)
no.chimera <- removeBimeraDenovo(seqtab, multithread=TRUE)
saveRDS(no.chimera, "/home/bryan/ERA/data/MiSeq/20170417_run1/v4bac_lib1_demult/lib1_seqtab.rds") 
```

# Checkpoint, possible to read file an continue from here

```{r}
taxa <- assignTaxonomy(no.chimera, "/home/bryan/PennRhiz/data/databases/rdp_train_set_14.fa.gz")
colnames(taxa) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")
unname(head(taxa))
```

# Make OTU names more readable

```{r}
otu.seqs <- colnames(no.chimera)
otu.names <- paste("OTU", 1:length(otu.seqs), sep=".")
otu.df <- data.frame(otu.names, otu.seqs)
otu.seqtab = no.chimera
colnames(otu.seqtab) <- otu.df$otu.names
rownames(taxa) <- otu.df$otu.names
```

# Create phyloseq object
```{r}
library(phyloseq)
sample_data = import_qiime_sample_data("/home/bryan/ERA/ERA_MappingFile.txt")

ps <- phyloseq(otu_table(otu.seqtab, taxa_are_rows=FALSE), 
               sample_data(sample_data), 
               tax_table(taxa))
saveRDS(ps, file = "/home/bryan/ERA/data/MiSeq/20170417_run1/v4bac_lib1_demult/ERA_physeq.rds")
```

```{r}
ps
```

